integration/spark/src/test/java/io/openlineage/spark/agent/lifecycle/RecursiveMatcher.java

```diff
@@ -0,0 +1,43 @@
+package io.openlineage.spark.agent.lifecycle;
+
+import io.openlineage.spark.agent.client.OpenLineageClient;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import org.assertj.core.api.Condition;
+
+public class RecursiveMatcher extends Condition<Map<String, Object>> {
```

Comment:
If we're going to extract this out, can we name it something more meaningful and descriptive? Something like `MatchesMapRecursively`